adding hwloc-gpu variant

MatthewLieber opened the pull request:
Adds a variant that by default fixes issues where building on nodes with gpus and then trying to run on nodes with out them will fail.

Thanks to Ron Rahaman @PACE/GaTech for the report

MatthewLieber committed:
adding hwloc-gpu variant

MatthewLieber committed:
rename to hwloc-graphics to avoid confusion

natshineman reviewed (dismissed):
This should be good to merge.

MatthewLieber commented:
Can the checks please be rerun?  It looks like it might be stuck as it has been pending all weekend.

MatthewLieber commented:
@alalazo All the checks have passed can we please get this merged in?

alalazo commented on var/spack/repos/builtin/packages/mvapich2/package.py:
```diff
@@ -109,6 +109,7 @@ class Mvapich2(AutotoolsPackage):
     )
 
     variant("hwlocv2", default=False, description="Builds mvapich2 with hwloc v2")
+    variant("hwloc-graphics", default=False, description="Enables hwloc graphics")
```
Can we do:
```suggestion
    variant("hwloc_graphics", default=False, description="Enables hwloc graphics")
```
for consistency with other variants?

alalazo commented on var/spack/repos/builtin/packages/mvapich2/package.py:
```diff
@@ -430,7 +431,10 @@ def configure_args(self):
         else:
             args.append("--disable-cuda")
-
+        if "~hwloc-graphics" in self.spec:
```
Do we need to enable anything explicitly if this variant is active?

MatthewLieber replied:
Nothing needs to be done explicitly if they activate the variant.

MatthewLieber committed:
change variant name to be more consistent

haampie committed:
gha bootstrap-dev-rhel8: configure git safe.directory (#37702)

git has been updated to something more recent

scheibelp committed:
Requirements and preferences should not define (non-git) versions (#37687)

Ensure that requirements `packages:*:require:@x` and preferences `packages:*:version:[x]`
fail concretization when no version defined in the package satisfies `x`. This always holds
except for git versions -- they are defined on the fly.

haampie committed:
Revert "hdf5: fix showconfig (#34920)" (#37707)

This reverts commit 192e564e26e919169aa926e92af7bd9a28cbb4f1.

haampie committed:
gha rhel8-platform-python: configure git safe.directory (#37708)

kwryankrattiger committed:
CI: Expand E4S ROCm stack to include missing DaV packages (#36843)

* CI: Expand E4S ROCm stack to include missing DaV packages

Ascent: Fixup for VTK-m with Kokkos backend

* DaV SDK: Removed duplicated openmp variant for ascent

* Drop visit and add conflict for Kokkos

* E4S: Drop ascent from CUDA builds

johnwparent committed:
Windows: fix MSVC version handling (#37711)

MSVC compiler logic was using string parsing to extract version
from compiler spec, which was fragile. This broke in #37572, so has
been fixed and made more robust by using attribute access.

greenc-FNAL committed:
libxpm package: fix RHEL8 build with libintl (#37713)

Set LDFLAGS rather than LDLIBS

alalazo committed:
Fix `spack find` not able to display version ranges in compilers (#37715)

tldahlgren committed:
emacs: convert to new stand-alone test process (#37725)

panda1100 committed:
Add recently added Spack Docker Images to documentation (#37732)

Signed-off-by: Yoshiaki Senda <yoshiaki@live.it>

scheibelp committed:
Tk/Tcl packages: speed up file search (#35902)

sethrj committed:
celeritas: new version 0.2.2 (#37731)

* celeritas: new version 0.2.2

* [@spackbot] updating style on behalf of sethrj

tldahlgren committed:
spack test: fix stand-alone test suite status reporting (#37602)

* Fix reporting of packageless specs as having no tests

* Add test_test_output_multiple_specs with update to simple-standalone-test (and tests)

* Refactored test status summary; added more tests or checks

haampie committed:
libxcb: depend on python, remove releases that need python 2 (#37698)

jmcarcell committed:
whizard: build newer versions in parallel (#37422)

scottwittenburg committed:
gitlab ci: no copy-only pipelines w/ deprecated config (#37720)

Make it clear that copy-only pipelines are not supported while still
using the deprecated ci config format. Also ensure that the deprecated
stack does not fail on spack pipelines for tags.

alalazo committed:
Limit deepcopy to just the initial "all" section (#37718)

Modifications:
- [x] Limit the scope of the deepcopy when initializing module file writers

haampie committed:
lmod: fix build, bump patch version (#37744)

bvanessen committed:
Added version 1.3.1 (#37735)

Commit:
Add aws-plcuster[-aarch64] stacks  (#37627)

Add aws-plcuster[-aarch64] stacks.  These stacks build packages defined in
https://github.com/spack/spack-configs/tree/main/AWS/parallelcluster

They use a custom container from https://github.com/spack/gitlab-runners which
includes necessary ParallelCluster software to link and build as well as an
upstream spack installation with current GCC and dependencies.

Intel and ARM software is installed and used during the build stage but removed
from the buildcache before the signing stage.

Files `configs/linux/{arch}/ci.yaml` select the necessary providers in order to
build for specific architectures (icelake, skylake, neoverse_{n,v}1).

tldahlgren committed:
maintainers: switch from list to directive (#37752)

scheibelp committed:
Bugfix: allow preferred new versions from externals (#37747)

alalazo committed:
Bump Spack version on develop to 0.21.0.dev0 (#37760)

greenc-FNAL committed:
[xrootd] New variants, new version, improve build config (#37682)

* Add FNAL Spack team to maintainers

* New variants and configuration improvements

* Version dependent "no-systemd" patches.

* New variants `client_only`, and `davix`

* Better handling of `cxxstd` for different versions, including
  improved patching and CMake options.

* Version-specific CMake requirements.

* Better version-specific handling of `openssl` dependency.

* `py-setuptools` required for `+python` build.

* Specific enable/disable of CMake options and use of
  `-DFORCE_ENABLED=TRUE` to prevent unwanted/non-portable activation
  of features.

* Better handling of `+python` configuration.

* New version 5.5.5

CamStan committed:
unifyfs: new v1.1 release (#37756)

Add v1.1 release
Update mochi-margo dependency compatible versions
Update version range of libfabric conflict

glennpj committed:
Cran updates (#37296)

* add version 1.28.3 to r-hexbin
* add version 5.0-1 to r-hmisc
* add version 0.5.5 to r-htmltools
* add version 1.6.2 to r-htmlwidgets
* add version 1.4.2 to r-igraph
* add version 0.42.19 to r-imager
* add version 1.0-5 to r-inum
* add version 0.9-14 to r-ipred
* add version 1.3.2 to r-irkernel
* add version 2.2.0 to r-janitor
* add version 0.1-10 to r-jpeg
* add version 1.2.2 to r-jsonify
* add version 0.9-32 to r-kernlab
* add version 1.7-2 to r-klar
* add version 1.42 to r-knitr
* add version 1.14.0 to r-ks
* add version 2.11.0 to r-labelled
* add version 1.7.2.1 to r-lava
* add version 0.6-15 to r-lavaan
* add version 2.1.2 to r-leaflet
* add version 2.9-0 to r-lfe
* add version 1.1.6 to r-lhs
* add version 1.1-33 to r-lme4
* add version 1.5-9.7 to r-locfit
* add version 0.4.3 to r-log4r
* add version 5.6.18 to r-lpsolve
* add version 0.2-11 to r-lwgeom
* add version 2.7.4 to r-magick
* add version 1.22.1 to r-maldiquant
* add version 1.2.11 to r-mapproj
* add version 1.6 to r-markdown
* add version 7.3-59 to r-mass
* add version 1.5-4 to r-matrix
* add version 0.63.0 to r-matrixstats
* add version 4.2-3 to r-memuse
* add version 4.0-0 to r-metafor
* add version 1.8-42 to r-mgcv
* add version 3.15.0 to r-mice
* add version 0.4-5 to r-mitml
* add version 2.0.0 to r-mixtools
* add version 0.1.11 to r-modelr
* add version 1.4-23 to r-multcomp
* add version 0.1-9 to r-multcompview
* add version 0.1-13 to r-mutoss
* add version 1.18.1 to r-network
* add version 3.3.4 to r-nleqslv
* add version 3.1-162 to r-nlme
* add version 0.26 to r-nmf
* add version 0.60-17 to r-np
* add version 4.2.5.2 to r-openxlsx
* add version 2022.11-16 to r-ordinal
* add version 0.6.0.8 to r-osqp
* add version 0.9.1 to r-packrat
* add version 1.35.0 to r-parallelly
* add version 1.3-13 to r-party
* add version 1.2-20 to r-partykit
* add version 1.7-0 to r-pbapply
* add version 0.3-9 to r-pbdzmq
* add version 1.2 to r-pegas
* add version 1.5-1 to r-phytools
* add version 1.9.0 to r-pillar
* add version 1.4.0 to r-pkgbuild
* add version 2.1.0 to r-pkgcache
* add version 0.5.0 to r-pkgdepends
* add version 2.0.7 to r-pkgdown
* add version 1.3.2 to r-pkgload
* add version 0.1-8 to r-png
* add version 1.1.22 to r-polspline
* add version 1.0.1 to r-pool
* add version 1.4.1 to r-posterior
* add version 3.8.1 to r-processx
* add version 2023.03.31 to r-prodlim
* add version 1.0-12 to r-proj4
* add version 2.5.0 to r-projpred
* add version 0.1.6 to r-pryr
* add version 1.7.5 to r-ps
* add version 1.0.1 to r-purrr
* add version 1.3.2 to r-qqconf
* add version 0.25.5 to r-qs
* add version 1.60 to r-qtl
* add version 0.4.22 to r-quantmod
* add version 5.95 to r-quantreg
* add version 0.7.8 to r-questionr
* add version 1.2.5 to r-ragg
* add version 0.15.1 to r-ranger
* add version 3.6-20 to r-raster
* add version 2.2.13 to r-rbibutils
* add version 1.0.10 to r-rcpp
* add version 0.12.2.0.0 to r-rcpparmadillo
* add version 0.1.7 to r-rcppde
* add version 0.3.13 to r-rcppgsl
* add version 1.98-1.12 to r-rcurl
* add version 1.2-1 to r-rda
* add version 2.1.4 to r-readr
* add version 1.4.2 to r-readxl
* add version 1.0.6 to r-recipes
* add version 1.1.6 to r-repr
* add version 1.2.16 to r-reproducible
* add version 0.3.0 to r-require
* add version 1.28 to r-reticulate
* add version 2.0.7 to r-rfast
* add version 1.6-6 to r-rgdal
* add version 0.6-2 to r-rgeos
* add version 1.1.3 to r-rgl
* add version 0.2.18 to r-rinside
* add version 4-14 to r-rjags
* add version 1.3-1.8 to r-rjsonio
* add version 2.21 to r-rmarkdown
* add version 0.9-2 to r-rmpfr
* add version 0.7-1 to r-rmpi
* add version 6.6-0 to r-rms
* add version 0.10.25 to r-rmysql
* add version 0.8.7 to r-rncl
* add version 2.4.11 to r-rnexml
* add version 0.95-1 to r-robustbase
* add version 1.3-20 to r-rodbc
* add version 7.2.3 to r-roxygen2
* add version 1.4.5 to r-rpostgres
* add version 0.7-5 to r-rpostgresql
* add version 0.8.29 to r-rsconnect
* add version 0.4-15 to r-rsnns
* add version 2.3.1 to r-rsqlite
* add version 0.7.2 to r-rstatix
* add version 1.1.2 to r-s2
* add version 0.4.5 to r-sass
* add version 0.1.9 to r-scatterpie
* add version 0.3-43 to r-scatterplot3d
* add version 3.2.4 to r-scs
* add version 1.6-4 to r-segmented
* add version 4.2-30 to r-seqinr
* add version 0.26 to r-servr
* add version 4.3.0 to r-seurat
* add version 1.0-12 to r-sf
* add version 0.4.2 to r-sfheaders
* add version 1.1-15 to r-sfsmisc
* add version 1.7.4 to r-shiny
* add version 1.9.0 to r-signac
* add version 1.6.0.3 to r-smoof
* add version 0.1.7-1 to r-sourcetools
* add version 1.6-0 to r-sp
* add version 1.3-0 to r-spacetime
* add version 7.3-16 to r-spatial
* add version 2.0-0 to r-spatialeco
* add version 1.2-8 to r-spatialreg
* add version 3.0-5 to r-spatstat
* add version 3.0-1 to r-spatstat-data
* add version 3.1-0 to r-spatstat-explore
* add version 3.1-0 to r-spatstat-geom
* add version 3.1-0 to r-spatstat-linnet
* add version 3.1-4 to r-spatstat-random
* add version 3.0-1 to r-spatstat-sparse
* add version 3.0-2 to r-spatstat-utils
* add version 2.2.2 to r-spdata
* add version 1.2-8 to r-spdep
* add version 0.6-1 to r-stars
* add version 1.5.0 to r-statmod
* add version 4.8.0 to r-statnet-common
* add version 1.7.12 to r-stringi
* add version 1.5.0 to r-stringr
* add version 1.9.1 to r-styler
* add version 3.5-5 to r-survival
* add version 1.5-4 to r-tclust
* add version 1.7-29 to r-terra
* add version 3.1.7 to r-testthat
* add version 1.1-2 to r-th-data
* add version 1.2 to r-tictoc
* add version 1.3.2 to r-tidycensus
* add version 1.2.3 to r-tidygraph
* add version 1.3.0 to r-tidyr
* add version 2.0.0 to r-tidyverse
* add version 0.2.0 to r-timechange
* add version 0.45 to r-tinytex
* add version 0.4.1 to r-triebeard
* add version 1.0-9 to r-truncnorm
* add version 0.10-53 to r-tseries
* add version 0.8-1 to r-units
* add version 4.3.0 to r-v8
* add version 1.4-11 to r-vcd
* add version 1.14.0 to r-vcfr
* add version 0.6.2 to r-vctrs
* add version 1.1-8 to r-vgam
* add version 0.4.0 to r-vioplot
* add version 1.6.1 to r-vroom
* add version 1.72-1 to r-wgcna
* add version 0.4.1 to r-whisker
* add version 0.7.2 to r-wk
* add version 0.39 to r-xfun
* add version 1.7.5.1 to r-xgboost
* add version 1.0.7 to r-xlconnect
* add version 3.99-0.14 to r-xml
* add version 0.13.1 to r-xts
* add version 2.3.7 to r-yaml
* add version 2.3.0 to r-zip
* add version 1.8-12 to r-zoo
* r-bigmem: dependency on uuid
* r-bio3d: dependency on zlib
* r-devtools: dependency cleanup
* r-dose: dependency cleanup
* r-dss: dependency cleanup
* r-enrichplot: dependency cleanup
* r-fgsea: dependency cleanup
* r-geor: dependency cleanup
* r-ggridges: dependency cleanup
* r-lobstr: dependency cleanup
* r-lubridate: dependency cleanup
* r-mnormt: dependency cleanup
* r-sctransform: version format correction
* r-seuratobject: dependency cleanup
* r-tidyselect: dependency cleanup
* r-tweenr: dependency cleanup
* r-uwot: dependency cleanup
* new package: r-clock
* new package: r-conflicted
* new package: r-diagram
* new package: r-doby
* new package: r-httr2
* new package: r-kableextra
* new package: r-mclogit
* new package: r-memisc
* new package: r-spatstat-model
* r-rmysql: use mariadb-client
* r-snpstats: add zlib dependency
* r-qs: add zstd dependency
* r-rcppcnpy: add zlib dependency
* black reformatting
* Revert "r-dose: dependency cleanup"
  This reverts commit 4c8ae8f5615ee124fff01ce43eddd3bb5d06b9bc.
* Revert "r-dss: dependency cleanup"
  This reverts commit a6c5c15c617a9a688fdcfe2b70c501c3520d4706.
* Revert "r-enrichplot: dependency cleanup"
  This reverts commit 65e116c18a94d885bc1a0ae667c1ef07d1fe5231.
* Revert "r-fgsea: dependency cleanup"
  This reverts commit ffe2cdcd1f73f69d66167b941970ede0281b56d7.
* r-rda: this package is back in CRAN
* r-sctransform: fix copyright
* r-seurat: fix copyright
* r-seuratobject: fix copyright
* Revert "add version 6.0-94 to r-caret"
  This reverts commit 236260597de97a800bfc699aec1cd1d0e3d1ac60.
* add version 6.0-94 to r-caret
* Revert "add version 1.8.5 to r-emmeans"
  This reverts commit 64a129beb0bd88d5c88fab564cade16c03b956ec.
* add version 1.8.5 to r-emmeans
* Revert "add version 5.0-1 to r-hmisc"
  This reverts commit 517643f4fd8793747365dfcfc264b894d2f783bd.
* add version 5.0-1 to r-hmisc
* Revert "add version 1.42 to r-knitr"
  This reverts commit 2a0d9a4c1f0ba173f7423fed59ba725bac902c37.
* add version 1.42 to r-knitr
* Revert "add version 1.6 to r-markdown"
  This reverts commit 4b5565844b5704559b819d2e775fe8dec625af99.
* add version 1.6 to r-markdown
* Revert "add version 0.26 to r-nmf"
  This reverts commit 4c44a788b17848f2cda67b32312a342c0261caec.
* add version 0.26 to r-nmf
* Revert "add version 2.3.1 to r-rsqlite"
  This reverts commit 5722ee2297276e4db8beee461d39014b0b17e420.
* add version 2.3.1 to r-rsqlite
* Revert "add version 1.0-12 to r-sf"
  This reverts commit ee1734fd62cc02ca7a9359a87ed734f190575f69.
* add version 1.0-12 to r-sf
* fix syntax error

glennpj committed:
update R: add version 4.3.0 (#37090)

kaanolgu committed:
[babelstream] FIX: maintainers list missing a comma (#37761)

michaelkuhn committed:
harfbuzz: add 7.3.0 (#37753)

Do not prefer 5.3.1 anymore since it does not build with newer compilers. Linux distributions have also moved to 6.x and newer.

vlopezh committed:
dlb: add v3.3.1 (#37759)

climbfuji committed:
Remove references to gmake executable, only use make (#37280)

vucoda committed:
Update py-jedi 0.17.2 checksum in package.py (#37700)

Pypi for jedi-0.17.2.tar.gz does not match the package.py file.

https://pypi.org/project/jedi/0.17.2/#copy-hash-modal-60e943e3-1192-4b12-90e2-4d639cb5b4f7

meyersbs committed:
New package py-scikit-tensor-py3 (#37706)

meyersbs committed:
New package py-tensorly (#37705)

abhamidipati-msft committed:
update package nccl version 2.17, 2.18 (#37721)

meyersbs committed:
Update py-nltk (#37703)

* Update py-nltk

* [@spackbot] updating style on behalf of meyersbs

* Update var/spack/repos/builtin/packages/py-nltk/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

---------

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

hyoklee committed:
py-jarvis-util: add a new package (#37729)

* py-jarvis-util: add a new package

* Apply suggestions from code review

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

---------

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

alecbcs committed:
rdkit: add v2023_03_1 (#37529)

alecbcs committed:
libjwt: add v1.15.3 (#37521)

alecbcs committed:
amqp-cpp: add v4.3.24 (#37504)

alecbcs committed:
roctracer-dev-api: add v5.5.0 (#37484)

alecbcs committed:
fzf: add v0.40.0 and refactor package (#37569)

* fzf: add v0.40.0 and refactor package
* Remove unused imports

eugeneswalker committed:
legion +rocm: apply patch for --offload-arch (#37775)

* legion +rocm: apply patch for --offload-arch

* constrain to latest version

pghysels committed:
Add STRUMPACK versions 7.1.2 and 7.1.3 (#37779)

cyrush committed:
add conduit 0.8.8 release (#37776)

balay committed:
llvm: add version 16.0.3 (#37472)

becker33 committed:
compiler specs: do not print '@=' when clear from context (#37787)

Ensure that spack compiler add/find/list and lists of concrete specs
print the compiler effectively as {compiler.name}{@compiler.version}.

Co-authored-by: Harmen Stoppels <harmenstoppels@gmail.com>

alalazo committed:
lmod: allow core compiler to be specified with a version range (#37789)

Use CompilerSpec with satisfies instead of string equality tests

Co-authored-by: Harmen Stoppels <harmenstoppels@gmail.com>

tldahlgren committed:
formetis: converted to new stand-alone test process (#37785)

tldahlgren committed:
fortrilinos: convert to new stand-alone test process (#37783)

alalazo committed:
archspec: fix entry in the JSON file (#37793)

robertu94 committed:
updates for the libpressio ecosystem (#37764)

* updates for the libpressio ecosystem

* [@spackbot] updating style on behalf of robertu94

* style fix: remove FIXME

---------

Co-authored-by: Robert Underwood <runderwood@anl.gov>
Co-authored-by: eugeneswalker <eugenesunsetwalker@gmail.com>

Thomas-Ulrich committed:
hipsycl: add v0.9.4 (#37247)

haampie committed:
spack env create: generate a view when newly created env has concrete specs (#37799)

mwkrentel committed:
intel-xed: add version 2023.04.16 (#37582)

* intel-xed: add version 2023.04.16
 1. add version 2023.04.16
 2. adjust the mbuild resource to better match the xed version at the time
 3. replace three conflicts() with one new requires() for x86_64 target
 4. add patch for libxed-ild for some new avx512 instructions
    * [@spackbot] updating style on behalf of mwkrentel
    * Fix the build for 2023.04.16.  XED requires its source directory to be exactly 'xed', so add a symlink.
 5. move the mbuild resource up one level, xed wants it to be in the same directory as the xed source dir
 6. deprecate 10.2019.03
    * semantic style fix: add OSError to except
    * [@spackbot] updating style on behalf of mwkrentel

---------

Co-authored-by: mwkrentel <mwkrentel@users.noreply.github.com>

perarnau committed:
aml: v0.2.1 (#37621)

* aml: v0.2.1

* add version 0.2.1
* fix hip variant bug

* [fix] pkgconf required for all builds

On top of needing pkgconf for autoreconf builds, the release configure
scripts needs pkgconf do detect dependencies if any of the hwloc, ze, or
opencl variants are active.

* Remove deprecation for v0.2.0 based on PR advise.

haampie committed:
fix binutils dep of spack itself (#37738)

vsoch committed:
Automated deployment to update package flux-security 2023-05-16 (#37696)

Co-authored-by: github-actions <github-actions@users.noreply.github.com>

joseeroman committed:
New patch release SLEPc 3.19.1 (#37675)

* New patch release SLEPc 3.19.1

* py-slepc4py: add explicit dependency on py-numpy

manuelakuhn committed:
py-alabaster: add 0.7.13 (#37798)

manuelakuhn committed:
py-anyio: add 3.6.2 (#37796)

manuelakuhn committed:
libtiff: disable use of sphinx (#37803)

kwryankrattiger committed:
Mochi-Margo: Add patch for pthreads detection (#36109)

manuelakuhn committed:
py-argcomplete: add 3.0.8 (#37797)

* py-argcomplete: add 3.0.8

* Update var/spack/repos/builtin/packages/py-argcomplete/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

* [@spackbot] updating style on behalf of manuelakuhn

---------

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

tgamblin committed:
Update `CHANGELOG.md` for v0.20.0

aprokop committed:
arborx: new version 1.4 (#37809)

Jordan474 committed:
llvm: Fix hwloc@1 and hwloc@:2.3 compatibility (#35387)

tldahlgren committed:
swig: convert to new stand-alone test process (#37786)

simonLeary42 committed:
deleted package.py better error message (#37814)

adds the namespace to the exception object's string representation

eugeneswalker committed:
tau %oneapi: -Wno-error=implicit-function-declaration (#37829)

jmcarcell committed:
gaudi: fix issue with fmt::format (#37810)

Co-authored-by: jmcarcell <jmcarcell@users.noreply.github.com>

adamjstewart committed:
azcopy: add new package (#37693)

adamjstewart committed:
py-fiona: add v1.9.4 (#37780)

adamjstewart committed:
py-huggingface-hub: add v0.14.1, cli variant (#37815)

genric committed:
py-dask: add v2023.4.1 (#37550)

* py-dask: add v2023.4.1

* address review comments

manuelakuhn committed:
py-asttokens: add 2.2.1 (#37816)

snehring committed:
py-pastml: adding version for compatibility with py-topiary-asr (#37828)

jmcarcell committed:
dd4hep: restrict podio versions (#37699)

Co-authored-by: jmcarcell <jmcarcell@users.noreply.github.com>

jmcarcell committed:
gaudi: add gaudi to LD_LIBRARY_PATH (#37821)

Co-authored-by: jmcarcell <jmcarcell@users.noreply.github.com>

jmcarcell committed:
dd4hep: add LD_LIBRARY_PATH for plugins for Gaudi (#37824)

Co-authored-by: jmcarcell <jmcarcell@users.noreply.github.com>

tgamblin committed:
libgcrypt: patch 1.10.2 on macos (#37844)

macOS doesn't have `getrandom`, and 1.10.2 fails to compile because of this.

There's an upstream fix at https://dev.gnupg.org/T6442 that will be in the next
`libgcrypt` release, but the patch is available now.

Commit:
hpcg: apply patch with openmp pragma changes for intel and oneapi compilers (#37856)

Signed-off-by: Andrey Parfenov <andrey.parfenov@intel.com>

snehring committed:
usalign: new package (#37646)

* usalign: adding new package
* usalign: updating shasum, adding note about distribution

glennpj committed:
Bioc updates (#37297)

* add version 1.48.0 to bioconductor package r-a4
* add version 1.48.0 to bioconductor package r-a4base
* add version 1.48.0 to bioconductor package r-a4classif
* add version 1.48.0 to bioconductor package r-a4core
* add version 1.48.0 to bioconductor package r-a4preproc
* add version 1.48.0 to bioconductor package r-a4reporting
* add version 1.54.0 to bioconductor package r-absseq
* add version 1.30.0 to bioconductor package r-acde
* add version 1.78.0 to bioconductor package r-acgh
* add version 2.56.0 to bioconductor package r-acme
* add version 1.70.0 to bioconductor package r-adsplit
* add version 1.72.0 to bioconductor package r-affxparser
* add version 1.78.0 to bioconductor package r-affy
* add version 1.76.0 to bioconductor package r-affycomp
* add version 1.58.0 to bioconductor package r-affycontam
* add version 1.72.0 to bioconductor package r-affycoretools
* add version 1.48.0 to bioconductor package r-affydata
* add version 1.52.0 to bioconductor package r-affyilm
* add version 1.70.0 to bioconductor package r-affyio
* add version 1.76.0 to bioconductor package r-affyplm
* add version 1.46.0 to bioconductor package r-affyrnadegradation
* add version 1.48.0 to bioconductor package r-agdex
* add version 3.32.0 to bioconductor package r-agilp
* add version 2.50.0 to bioconductor package r-agimicrorna
* add version 1.32.0 to bioconductor package r-aims
* add version 1.32.0 to bioconductor package r-aldex2
* add version 1.38.0 to bioconductor package r-allelicimbalance
* add version 1.26.0 to bioconductor package r-alpine
* add version 2.62.0 to bioconductor package r-altcdfenvs
* add version 2.24.0 to bioconductor package r-anaquin
* add version 1.28.0 to bioconductor package r-aneufinder
* add version 1.28.0 to bioconductor package r-aneufinderdata
* add version 1.72.0 to bioconductor package r-annaffy
* add version 1.78.0 to bioconductor package r-annotate
* add version 1.62.0 to bioconductor package r-annotationdbi
* add version 1.24.0 to bioconductor package r-annotationfilter
* add version 1.42.0 to bioconductor package r-annotationforge
* add version 3.8.0 to bioconductor package r-annotationhub
* add version 3.30.0 to bioconductor package r-aroma-light
* add version 1.32.0 to bioconductor package r-bamsignals
* add version 2.16.0 to bioconductor package r-beachmat
* add version 2.60.0 to bioconductor package r-biobase
* add version 2.8.0 to bioconductor package r-biocfilecache
* add version 0.46.0 to bioconductor package r-biocgeneric
* add version 1.10.0 to bioconductor package r-biocio
* add version 1.18.0 to bioconductor package r-biocneighbors
* add version 1.34.0 to bioconductor package r-biocparallel
* add version 1.16.0 to bioconductor package r-biocsingular
* add version 2.28.0 to bioconductor package r-biocstyle
* add version 3.17.1 to bioconductor package r-biocversion
* add version 2.56.0 to bioconductor package r-biomart
* add version 1.28.0 to bioconductor package r-biomformat
* add version 2.68.0 to bioconductor package r-biostrings
* add version 1.48.0 to bioconductor package r-biovizbase
* add version 1.10.0 to bioconductor package r-bluster
* add version 1.68.0 to bioconductor package r-bsgenome
* add version 1.36.0 to bioconductor package r-bsseq
* add version 1.42.0 to bioconductor package r-bumphunter
* add version 2.66.0 to bioconductor package r-category
* add version 2.30.0 to bioconductor package r-champ
* add version 2.32.0 to bioconductor package r-champdata
* add version 1.50.0 to bioconductor package r-chipseq
* add version 4.8.0 to bioconductor package r-clusterprofiler
* add version 1.36.0 to bioconductor package r-cner
* add version 1.32.0 to bioconductor package r-codex
* add version 2.16.0 to bioconductor package r-complexheatmap
* add version 1.74.0 to bioconductor package r-ctc
* add version 2.28.0 to bioconductor package r-decipher
* add version 0.26.0 to bioconductor package r-delayedarray
* add version 1.22.0 to bioconductor package r-delayedmatrixstats
* add version 1.40.0 to bioconductor package r-deseq2
* add version 1.46.0 to bioconductor package r-dexseq
* add version 1.42.0 to bioconductor package r-dirichletmultinomial
* add version 2.14.0 to bioconductor package r-dmrcate
* add version 1.74.0 to bioconductor package r-dnacopy
* add version 3.26.0 to bioconductor package r-dose
* add version 2.48.0 to bioconductor package r-dss
* add version 3.42.0 to bioconductor package r-edger
* add version 1.20.0 to bioconductor package r-enrichplot
* add version 2.24.0 to bioconductor package r-ensembldb
* add version 1.46.0 to bioconductor package r-exomecopy
* add version 2.8.0 to bioconductor package r-experimenthub
* add version 1.26.0 to bioconductor package r-fgsea
* add version 2.72.0 to bioconductor package r-gcrma
* add version 1.36.0 to bioconductor package r-gdsfmt
* add version 1.82.0 to bioconductor package r-genefilter
* add version 1.36.0 to bioconductor package r-genelendatabase
* add version 1.72.0 to bioconductor package r-genemeta
* add version 1.78.0 to bioconductor package r-geneplotter
* add version 1.22.0 to bioconductor package r-genie3
* add version 1.36.0 to bioconductor package r-genomeinfodb
* update r-genomeinfodbdata
* add version 1.36.0 to bioconductor package r-genomicalignments
* add version 1.52.0 to bioconductor package r-genomicfeatures
* add version 1.52.0 to bioconductor package r-genomicranges
* add version 2.68.0 to bioconductor package r-geoquery
* add version 1.48.0 to bioconductor package r-ggbio
* add version 3.8.0 to bioconductor package r-ggtree
* add version 2.10.0 to bioconductor package r-glimma
* add version 1.12.0 to bioconductor package r-glmgampoi
* add version 5.54.0 to bioconductor package r-globaltest
* update r-go-db
* add version 1.20.0 to bioconductor package r-gofuncr
* add version 2.26.0 to bioconductor package r-gosemsim
* add version 1.52.0 to bioconductor package r-goseq
* add version 2.66.0 to bioconductor package r-gostats
* add version 1.78.0 to bioconductor package r-graph
* add version 1.62.0 to bioconductor package r-gseabase
* add version 1.32.0 to bioconductor package r-gtrellis
* add version 1.44.0 to bioconductor package r-gviz
* add version 1.28.0 to bioconductor package r-hdf5array
* add version 1.72.0 to bioconductor package r-hypergraph
* add version 1.36.0 to bioconductor package r-illumina450probevariants-db
* add version 0.42.0 to bioconductor package r-illuminaio
* add version 1.74.0 to bioconductor package r-impute
* add version 1.38.0 to bioconductor package r-interactivedisplaybase
* add version 2.34.0 to bioconductor package r-iranges
* add version 1.60.0 to bioconductor package r-kegggraph
* add version 1.40.0 to bioconductor package r-keggrest
* add version 3.56.0 to bioconductor package r-limma
* add version 2.52.0 to bioconductor package r-lumi
* add version 1.76.0 to bioconductor package r-makecdfenv
* add version 1.78.0 to bioconductor package r-marray
* add version 1.12.0 to bioconductor package r-matrixgenerics
* add version 1.8.0 to bioconductor package r-metapod
* add version 2.46.0 to bioconductor package r-methylumi
* add version 1.46.0 to bioconductor package r-minfi
* add version 1.34.0 to bioconductor package r-missmethyl
* add version 1.80.0 to bioconductor package r-mlinterfaces
* add version 1.12.0 to bioconductor package r-mscoreutils
* add version 2.26.0 to bioconductor package r-msnbase
* add version 2.56.0 to bioconductor package r-multtest
* add version 1.38.0 to bioconductor package r-mzid
* add version 2.34.0 to bioconductor package r-mzr
* add version 1.62.0 to bioconductor package r-oligoclasses
* update r-org-hs-eg-db
* add version 1.42.0 to bioconductor package r-organismdbi
* add version 1.40.0 to bioconductor package r-pathview
* add version 1.92.0 to bioconductor package r-pcamethods
* update r-pfam-db
* add version 1.44.0 to bioconductor package r-phyloseq
* add version 1.62.0 to bioconductor package r-preprocesscore
* add version 1.32.0 to bioconductor package r-protgenerics
* add version 1.34.0 to bioconductor package r-quantro
* add version 2.32.0 to bioconductor package r-qvalue
* add version 1.76.0 to bioconductor package r-rbgl
* add version 2.40.0 to bioconductor package r-reportingtools
* add version 2.44.0 to bioconductor package r-rgraphviz
* add version 2.44.0 to bioconductor package r-rhdf5
* add version 1.12.0 to bioconductor package r-rhdf5filters
* add version 1.22.0 to bioconductor package r-rhdf5lib
* add version 1.76.0 to bioconductor package r-roc
* add version 1.28.0 to bioconductor package r-rots
* add version 2.16.0 to bioconductor package r-rsamtools
* add version 1.60.0 to bioconductor package r-rtracklayer
* add version 0.38.0 to bioconductor package r-s4vectors
* add version 1.8.0 to bioconductor package r-scaledmatrix
* add version 1.28.0 to bioconductor package r-scater
* add version 1.14.0 to bioconductor package r-scdblfinder
* add version 1.28.0 to bioconductor package r-scran
* add version 1.10.0 to bioconductor package r-scuttle
* add version 1.66.0 to bioconductor package r-seqlogo
* add version 1.58.0 to bioconductor package r-shortread
* add version 1.74.0 to bioconductor package r-siggenes
* add version 1.22.0 to bioconductor package r-singlecellexperiment
* add version 1.34.0 to bioconductor package r-snprelate
* add version 1.50.0 to bioconductor package r-snpstats
* add version 2.36.0 to bioconductor package r-somaticsignatures
* add version 1.12.0 to bioconductor package r-sparsematrixstats
* add version 1.40.0 to bioconductor package r-spem
* add version 1.38.0 to bioconductor package r-sseq
* add version 1.30.0 to bioconductor package r-summarizedexperiment
* add version 3.48.0 to bioconductor package r-sva
* add version 1.38.0 to bioconductor package r-tfbstools
* add version 1.22.0 to bioconductor package r-tmixclust
* add version 2.52.0 to bioconductor package r-topgo
* add version 1.24.0 to bioconductor package r-treeio
* add version 1.28.0 to bioconductor package r-tximport
* add version 1.28.0 to bioconductor package r-tximportdata
* add version 1.46.0 to bioconductor package r-variantannotation
* add version 3.68.0 to bioconductor package r-vsn
* add version 2.6.0 to bioconductor package r-watermelon
* add version 2.46.0 to bioconductor package r-xde
* add version 1.58.0 to bioconductor package r-xmapbridge
* add version 0.40.0 to bioconductor package r-xvector
* add version 1.26.0 to bioconductor package r-yapsa
* add version 1.26.0 to bioconductor package r-yarn
* add version 1.46.0 to bioconductor package r-zlibbioc
* Revert "add version 1.82.0 to bioconductor package r-genefilter"
  This reverts commit 1702071c6d3ab83b75e0fcf493622113957223bb.
* Revert "add version 0.38.0 to bioconductor package r-s4vectors"
  This reverts commit 58a7df23872ac8035fdac75c93cedf6dd7819dd0.
* add version 0.38.0 to bioconductor package r-s4vectors
* Revert "add version 1.28.0 to bioconductor package r-aneufinder"
  This reverts commit 0a1f59de6c17fe6aa62bcf807826f9d2023280e6.
* add version 1.28.0 to bioconductor package r-aneufinder
* Revert "add version 2.16.0 to bioconductor package r-beachmat"
  This reverts commit cd49fb8e4c84763a063f1b2d0b3bdc0179161568.
* add version 2.16.0 to bioconductor package r-beachmat
* Revert "add version 4.8.0 to bioconductor package r-clusterprofiler"
  This reverts commit 6e9a951cbe08042ca5100f47b8b1680cae26cc14.
* add version 4.8.0 to bioconductor package r-clusterprofiler
* Fix syntax error
* r-genefilter: add version 1.82.0
* new package: r-basilisk-utils
* new package: r-basilisk
* new package: r-densvis
* new package: r-dir-expiry
* r-affyplm: add zlib dependency
* r-cner: add zlib dependency
* r-mzr: add zlib dependency
* r-rhdf5filters: add zstd dependency
* r-shortread: add zlib dependency
* r-snpstats: add zlib dependency

---------

Co-authored-by: Tamara Dahlgren <35777542+tldahlgren@users.noreply.github.com>

manuelakuhn committed:
py-bleach: add 6.0.0 (#37846)

kjrstory committed:
New package: FDS (#37850)

glennpj committed:
fplo: remove myself as maintainer (#37861)

glennpj committed:
gurobi: remove myself as maintainer (#37865)

glennpj committed:
mumax: remove myself as maintainer (#37869)

glennpj committed:
itk: remove myself as maintainer (#37867)

glennpj committed:
reditools: remove myself as maintainer (#37871)

hyoklee committed:
daos: add a new package (#35649)

glennpj committed:
gatetools: remove myself as maintainer (#37863)

eugeneswalker committed:
e4s ci: add dealii (#32484)

glennpj committed:
opencv: remove myself from maintainers list (#37870)

glennpj committed:
heasoft: remove myself as maintainer (#37866)

glennpj committed:
bart: remove myself as maintainer (#37860)

glennpj committed:
Remove myself as maintainer of R packages (#37859)

* Remove myself as maintainer of R packages
  I will no longer have the time to properly maintain these packages.
* fix flake8 test for import

glennpj committed:
gate: remove myself as maintainer (#37862)

LRWeber committed:
Updated hash for latest maintenance release (2022.2.1) (#37842)

glennpj committed:
julia: remove myself from maintainers list (#37868)

hyoklee committed:
hdf5-vol-daos: add a new package (#35653)

* hdf5-vol-daos: add a new package
* hdf5-vol-daos: address @soumagne review

---------

Co-authored-by: Tamara Dahlgren <35777542+tldahlgren@users.noreply.github.com>

houjun committed:
Add async vol v1.6 (#37875)

simonpintarelli committed:
update nlcglib package (#37578)

haampie committed:
spack remove: fix traversal when user specs intersect (#37882)

drop unnecessary double loop over the matching user specs.

becker33 committed:
bugfix: env concretize after remove (#37877)

Commit:
Pcluster local buildcache (#37852)

* [pcluster pipeline] Use local buildcache instead of upstream spack

Spack currently does not relocate compiler references from upstream spack
installations. When using a buildcache we don't need an upstream spack.

* gcc needs to be installed via postinstall to get correct deps

* quantum-espresso@gcc@12.3.0 returns ICE on neoverse_{n,v}1

* Force gitlab to pull the new container

* Revert "Force gitlab to pull the new container"

This reverts commit 3af5f4cd88245138992deb2a46c17e6f85858d68.

Seems the gitlab version does not yet support "pull_policy" in .gitlab-ci.yml

* Gitlab keeps picking up wrong container. Renaming

* Update containers once more after failed build

willdunklin committed:
sensei: Allow Paraview 5.11 for sensei develop version (#37719)

Commit:
paraview (protobuf failure) #37437 (#37440)

When attempting to build paraview@5.10.1 using a recent Intel
compiler (Classic or OneAPI) or the IBM XL compiler, the build
fails if the version of protobuf used is > 3.18

nfurmento committed:
starpu: add release 1.4.1 (#37883)

adamjstewart committed:
py-lightly: add v1.4.6 (#37885)

adamjstewart committed:
py-rasterio: add v1.3.7 (#37886)

manuelakuhn committed:
py-babel: add 2.12.1 (#37818)

* py-babel: add 2.12.1

* Update var/spack/repos/builtin/packages/py-babel/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

---------

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

manuelakuhn committed:
py-attrs: add 23.1.0 (#37817)

* py-attrs: add 23.1.0

* Add missing dependency

manuelakuhn committed:
py-certifi: add 2023.5.7 (#37848)

manuelakuhn committed:
py-bottleneck: add 1.3.7 (#37847)

manuelakuhn committed:
py-bids-validator: add 1.11.0 (#37845)

manuelakuhn committed:
py-charset-normalizer: add 3.1.0 (#37880)

eirrgang committed:
Update for py-gmxapi for 0.4.1. (#37834)

* Update for py-gmxapi for 0.4.1.

* Note 0.4.1 hash from PyPI.
* Note relaxed dependencies for future versions.

* Update var/spack/repos/builtin/packages/py-gmxapi/package.py

adamjstewart committed:
Add macOS ML CI stacks (#36586)

* Add macOS ML CI stacks

* torchmeta is no longer maintained and requires ancient PyTorch

* Add MXNet

* update darwin aarch64 stacks

* add darwin-aarch64 scoped config.yaml

* remove unnecessary cleanup job

* fix specifications

* fix labels

* fix labels

* fix indent on tags specification

* no tags for trigger jobs

* try overriding tags in stack spack.yaml

* do not use CI_STACK_CONFIG_SCOPES

* incorporate config:install_tree:root: overrides and compiler defs

* copy relevant ci-scoped config settings directly into stack spack.yaml

* remove build-job-remove

* spack ci generate: add debug flag

* include cdash config directly in stack spack.yaml

* customize build-job script section to avoid absolute paths

* add any-job specification

* tags: use aarch64-macos instead of aarch64

* generate tags: use aarch64-macos instead of aarch64

* do not add morepadding

* use shared mirror; comment out known failures

* remove any-job

* nproc || true

* comment out specs failing due to bazel from cache codesign issue

---------

Co-authored-by: eugeneswalker <eugenesunsetwalker@gmail.com>

AlexanderRichert-NOAA committed:
Add shared variant to geos package (#37899)

Co-authored-by: alexrichert <alexrichert@gmail.com>

Commit:
[devito] Move to version 4.8.1 (#37915)

* [devito] Move to version 4.8.1

* Fix: Adding patch file

* Update var/spack/repos/builtin/packages/py-devito/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

* Update var/spack/repos/builtin/packages/py-devito/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

* Addressing @adamjstewart comments

---------

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

hyoklee committed:
fix(pmdk): add pkconfig as dependency (#37896)

Fix #37887 failure.

tldahlgren committed:
Bugfix/tests: write not append stand-alone test status (#37841)

xdelaruelle committed:
environment-modules: fix @main version requirements (#37807)

Some requirements for @main version of environment-modules were missing:

* python (to build ChangeLog documentation file)
* py-sphinx@1.0: (to build man-pages, etc)

Also adding gzip, which is now required to build ChangeLog.gz (which is
now shipped instead of ChangeLog).

Other versions are not requiring these tools (as documentation is
pre-built in dist tarball).

rbberger committed:
flecsi: disable cinch dependency for v1 release (#37857)

* flecsi: disable cinch dependency for v1 releases
* [@spackbot] updating style on behalf of rbberger

---------

Co-authored-by: rbberger <rbberger@users.noreply.github.com>

johnwparent committed:
Protobuf package: CMake fix for Windows build (#37926)

Qualify reference with namespace. A pending upstream PR will eventually
make this unnecessary, so the patch is only applied for 3.22. versions.

hainest committed:
Dyninst: add standalone test (#37876)

* Dyninst: add standalone test
* Add docstring with description
* Don't use join_path for builtin path objects
* Whitespace
* Update format of docstring

simonpintarelli committed:
sirius: add rocsolver/wannier90 (#37900)

* sirius: add rocsolver dependency for 7.5:
* add wannier90

hyoklee committed:
fix(dpdk): add a new version 23.03. (#37919)

Fix E4S GitLab CI issue #37887.

hyoklee committed:
fix(protobuf-c): set version bound for protobuf dependency (#37917)

Fix #37887 GitLab CI failure.

johnwparent committed:
Windows CI: add paraview deps nightly build (#37924)

Add a nightly job to attempt building all Paraview dependencies and
upload the results to cdash. This check doesn't affect the reported
build/test status of Spack. We are using this to monitor the state of
Windows support while working on more-robust checks (eventually the
Windows build will have to succeed to merge PRs to Spack).

eugeneswalker committed:
update ci ml darwin keypath (#37927)

johnwparent committed:
Windows: fix "spack build-env" (#37923)

"spack build-env" was not generating proper environment variable
definitions on Windows; this commit updates the generated commands
to succeed with batch/PowerShell.

Leonard-Anderson-NNL committed:
gradle:add 7.3 (#37928)

Co-authored-by: Cloud User <leonardanderson@leonardander001.hzterscemazurawp3xenxzahla.bx.internal.cloudapp.net>

A-N-Other committed:
fastp: add version 0.23.3, add build dependencies (#37931)

Co-authored-by: LMS Bioinformatics <bioinformatics@lms.mrc.ac.uk>

johnwparent committed:
Windows MSVC: do not set sdk version if installing sdk (#37930)

Note the win-sdk package is not installable and reports an error
which instructs the user how to add it. Without this fix, a
(more confusing) error occurs before this message can be generated.

hppritcha committed:
PMIx and PRRTe: disabled use of sphinx (#37750)

Related to https://github.com/spack/spack/pull/37717

No need to be rebuilding openmpi man pages and other docs in
spack as it almost always is used with release tarballs.

See #37717 for more details.

Signed-off-by: Howard Pritchard <howardp@lanl.gov>

Try2Code committed:
cdo: add 2.2.0 (#37244)

siko1056 committed:
Strip inactive maintainer. (#36048)

A-N-Other committed:
hyphy: add 2.5.51hf, update dependencies for switch to MPI (#37938)

Co-authored-by: LMS Bioinformatics <bioinformatics@lms.mrc.ac.uk>

A-N-Other committed:
minimap2: adding 2.26 (#37945)

Co-authored-by: LMS Bioinformatics <bioinformatics@lms.mrc.ac.uk>

A-N-Other committed:
picard: add 3.0.0, switch to java@17: (#37948)

Co-authored-by: LMS Bioinformatics <bioinformatics@lms.mrc.ac.uk>

kwryankrattiger committed:
CI: Use relative path in default script (#36649)

manuelakuhn committed:
py-chardet: add 5.1.0 (#37879)

* py-chardet: add 5.1.0

* Remove py-setuptools as run dependency

manuelakuhn committed:
py-beautifulsoup4: add 4.12.2 (#37820)

mlxd committed:
Update py-pennylane ecosystem to support v0.30.0 (#37763)

* Update PennyLane ecosystem for 0.30 release

* Update package dep versions

* Fix formatting

* Update dep versions

* Remove PL hard pin and rely on PLQ to define version

* Update var/spack/repos/builtin/packages/py-pennylane-lightning-kokkos/package.py

Co-authored-by: Vincent Michaud-Rioux <vincent.michaud-rioux@xanadu.ai>

* Convert pybind11 from build to link dep, and PL ver limit

---------

Co-authored-by: Vincent Michaud-Rioux <vincent.michaud-rioux@xanadu.ai>

manuelakuhn committed:
py-coverage: add 5.5 (#37922)

manuelakuhn committed:
py-contourpy: add 1.0.7 (#37914)

manuelakuhn committed:
py-coloredlogs: add 15.0.1 and py-humanfriendly: add 10.0 (#37905)

mschouler committed:
Update melissa build (#37609)

* Remove deprecated package

* Add up-to-date melissa builds

* Remove blank lines and FIXME comments

* Use directive syntax for maintainers and remove unnecessary comments

* Remove unused function

* Deprecate former melissa recipe

* Change melissa python package name

* Update setuptools and rapidjson dependencies versions

* Fix mypy error

* Restore rapidjson version

* Variant simplification

* Make variants lower case

* Deprecate former omitted version

* Make torch version consistent with requirement file

* Fix variants definition

* Fix style error

---------

Co-authored-by: Marc Schouler <marc.schouler@inria.fr>

manuelakuhn committed:
py-distro: add 1.8.0 (#37942)

manuelakuhn committed:
py-debugpy: add 1.6.7 (#37941)

manuelakuhn committed:
py-datalad-metadata-model: add 0.3.10 (#37937)

A-N-Other committed:
py-multiqc: add 1.14, bump dependencies (#37946)

Co-authored-by: LMS Bioinformatics <bioinformatics@lms.mrc.ac.uk>

AlexanderRichert-NOAA committed:
freetype: add pic and shared variants (#37898)

AlexanderRichert-NOAA committed:
Add shared and pic variants to libtiff (#37965)

* Add static-only option for libtiff

* update libtiff to add pic variant

* fix libtiff pic setting

A-N-Other committed:
py-cutadapt: add 4.4, 4.3, 4.2 versions (#37929)

* py-cutadapt: add 4.4, 4.3, 4.2 versions

* Update var/spack/repos/builtin/packages/py-cutadapt/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

* Update var/spack/repos/builtin/packages/py-cutadapt/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

* Update package.py

---------

Co-authored-by: LMS Bioinformatics <bioinformatics@lms.mrc.ac.uk>
Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

A-N-Other committed:
py-macs2: add 2.2.8, updated dependencies (#37944)

* py-macs2: add 2.2.8, updated dependencies

* Update package.py

---------

Co-authored-by: LMS Bioinformatics <bioinformatics@lms.mrc.ac.uk>

tldahlgren committed:
tests/amrex: convert to new stand-alone test process (#35698)

* amrex: convert to new stand-alone test process
* smoke->stand-alone

tldahlgren committed:
tests/bolt: convert to new stand-alone test process (#35695)

* bolt: convert to new stand-alone test process
* Remove redundant test_requires_compiler (so above directives)

tldahlgren committed:
archer: convert to new stand-alone test process (#35697)

IncubatorShokuhou committed:
copy namelist and xml to ./bin (#37933)

fspiga committed:
Adding NVIDIA HPC SDK 23.5 (#37913)

tldahlgren committed:
tests/darshan-runtime: convert to new stand-alone test process (#37838)

tmadlener committed:
whizard: Fix parallel build race condition (#37890)

annop-w committed:
acfl: add version 23.04.1 and fix checksums for 22.1 (#37908)

Checksums for 22.1 need updates due to an IT incident at developer.arm.com. The package tarballs needed to be recreated.

dependabot[bot] committed:
build(deps): bump actions/setup-python from 4.6.0 to 4.6.1 (#37894)

Bumps [actions/setup-python](https://github.com/actions/setup-python) from 4.6.0 to 4.6.1.
- [Release notes](https://github.com/actions/setup-python/releases)
- [Commits](https://github.com/actions/setup-python/compare/57ded4d7d5e986d7296eab16560982c6dd7c923b...bd6b4b6205c4dbad673328db7b31b7fab9e241c0)

---
updated-dependencies:
- dependency-name: actions/setup-python
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

tldahlgren committed:
tests/cmake: convert to new stand-alone test process (#37724)

xdelaruelle committed:
modules: improve default naming scheme (#37808)

Change default naming scheme for tcl modules for a more user-friendly
experience. 

Change from flat projection to "per software name" projection.

Flat naming scheme restrains module selection capabilities. The
`{name}/{version}...` scheme make possible to use user-friendly
mechanisms:

* implicit defaults (`module load git`)
* extended default (`module load git/2`)
* advanced version specifiers (`module load git@2:`)

Rombur committed:
Kokkos: remove unused variants (#37800)

Rombur committed:
kokkos-nvcc-wrapper: Remove unnecessary dependencies (#37794)

vsoch committed:
Automated deployment to update package flux-core 2023-05-04 (#37421)

Co-authored-by: github-actions <github-actions@users.noreply.github.com>

tldahlgren committed:
tests/upcxx: convert to new stand-alone test process (#37832)

Co-authored-by: Dan Bonachea <dobonachea@lbl.gov>

tldahlgren committed:
hdf: convert to new stand-alone test process (#37843)

tldahlgren committed:
gasnet: convert to new stand-alone test process (#35727)

tldahlgren committed:
patchelf: convert to new stand-alone test process (#37831)

tldahlgren committed:
tests/flibcpp: convert to new stand-alone test process (#37782)

alecbcs committed:
form: add v4.3.1 (#37984)

snehring committed:
Beast2: add v2.7.4, add javafx (#37419)

alecbcs committed:
gradle: add v8.1.1 (#37986)

alecbcs committed:
glog: add v0.6.0 (#37985)

alecbcs committed:
fasttransforms: add v0.6.2 (#37983)

alecbcs committed:
erlang: add v26.0 (#37982)

alecbcs committed:
erfa: add v2.0.0 (#37981)

alecbcs committed:
elfio: add v3.11 (#37980)

alecbcs committed:
double-conversion: add v3.3.0 (#37979)

alecbcs committed:
cyrus-sasl: add v2.1.28 (#37978)

adamjstewart committed:
py-segmentation-models-pytorch: add v0.3.3 (#37970)

alecbcs committed:
cpio: add v2.14 (#37975)

alecbcs committed:
cracklib: add v2.9.9 (#37976)

Thomas-Ulrich committed:
pumi: fix simmodsuite base variant, and mpi lib name (#37401)

alecbcs committed:
bedops: add v2.4.41 (#37972)

alecbcs committed:
cbc: add v2.10.9 (#37974)

tldahlgren committed:
tests/sip: convert to new stand-alone test process (#35693)

jedwards4b committed:
parallelio: add v2.6.0 release and ncint variant (#37805)

jmcarcell committed:
dd4hep: add patch to fix missing hits when using LCIO (#37854)

Co-authored-by: jmcarcell <jmcarcell@users.noreply.github.com>

tldahlgren committed:
Bugfix/tests: add slash to test log message (#37874)

jmcarcell committed:
xrootd: add patch for when libraries are installed in lib64 (#37858)

Co-authored-by: jmcarcell <jmcarcell@users.noreply.github.com>

hppritcha committed:
dealii - add platform-introspection variant (#37833)

This option is needed for DFT FE - or more accurately the check needs to
be checked off for a number of platforms or else the code doesn't work.

Signed-off-by: Howard Pritchard <howardp@lanl.gov>

tldahlgren committed:
berkeley-db: convert to new stand-alone test process (#35702)

amd-toolchain-support committed:
NAMD: add AVXTILES support (#37040)

vanderwb committed:
bbcp: fix cloning "master" version (#37802)

jfalcou committed:
Add package for KUMI tuple library (#37795)

hppritcha committed:
OPENMPI: disable use of sphinx (#37717)

Sphinx is used to build Open MPI manpages, etc. as part of the make dist
process to create release tarballs.  There should be no need/use to do
this within Spack.  Also some sites have older Sphinx installs which
aren't compatible with the needs of the Open MPI documentation.
For example, attempts to install openmpi@main fail at NERSC owing to
such a situation.

Since Spack normally is used to build from release tarballs, in which
the docs have already been installed, this should present no issues.

This configuration option will be ignored for older than 5.0.0 Open MPI releases.

Signed-off-by: Howard Pritchard <howardp@lanl.gov>

haampie committed:
fix InternalConcretizerError msg (#37791)

alkino committed:
catch2: new versions (#37742)

jgalarowicz committed:
survey: add v1.0.8 (#37385)

alecbcs committed:
hbase: add v2.5.3 (#37518)

Co-authored-by: Tamara Dahlgren <35777542+tldahlgren@users.noreply.github.com>

tldahlgren committed:
tests/arborx: convert to new stand-alone test process (#37778)


Co-authored-by: Andrey Prokopenko <andrey.prok@gmail.com>

lgarrison committed:
hdfview: add 3.1.4, 3.2.0, 3.3.0. Update dependencies. (#37745)

haampie committed:
Use @= in some packages (#37737)

Change the pattern @x.y:x.y.0 -> @=x.y

Co-authored-by: haampie <haampie@users.noreply.github.com>

annop-w committed:
armpl-gcc: add version 23.04.1 (#37907)

tldahlgren committed:
m4: convert to new stand-alone test process (#37723)

sxs-bot committed:
spectre: add v2023.05.16 (#37716)

Co-authored-by: sxs-bot <sxs-bot@users.noreply.github.com>

tldahlgren committed:
libsigsegv: convert to new stand-alone test process (#37691)

tldahlgren committed:
binutils: convert to new stand-alone test process (#37690)

albestro committed:
highfive: add newer versions + add git repo (#37635)

* add versions
* add develop pointing to git@master

vsoch committed:
add flux-security variant to flux-core (#37689)

This will build flux-security separately to have a flux-imp
that can be defined in a flux broker.toml. Note that the user
that wants a multi-user setup is recommended to create a view,
and then a system/broker.toml in flux config directory that
points to it.

Signed-off-by: vsoch <vsoch@users.noreply.github.com>
Co-authored-by: vsoch <vsoch@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump codecov/codecov-action from 3.1.3 to 3.1.4 (#37688)

Bumps [codecov/codecov-action](https://github.com/codecov/codecov-action) from 3.1.3 to 3.1.4.
- [Release notes](https://github.com/codecov/codecov-action/releases)
- [Changelog](https://github.com/codecov/codecov-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/codecov/codecov-action/compare/894ff025c7b54547a9a2a1e9f228beae737ad3c2...eaaf4bedf32dbdc6b720b63067d99c4d77d6047d)

---
updated-dependencies:
- dependency-name: codecov/codecov-action
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

vsoch committed:
Automated deployment to update package flux-core 2023-05-15 (#37684)

Co-authored-by: github-actions <github-actions@users.noreply.github.com>

michaelkuhn committed:
glib: add 2.76.3 and 2.76.2, update build systems (#37226)

This also converts the package to the new way of handling multiple build
systems.

Co-authored-by: michaelkuhn <michaelkuhn@users.noreply.github.com>

zzzoom committed:
nwchem: fix non-fftw fftw-api providers (#37442)

white238 committed:
amgx: add v2.2.0 and v2.3.0 (#37567)

snehring committed:
sentieon-genomics: replacing square brackets in help (#37622)

tldahlgren committed:
tests/biobambam2: convert to new stand-alone test process (#35696)

haampie committed:
Remove patchelf self-relocation (#33834)

tldahlgren committed:
sqlite: convert to new stand-alone test process (#37722)

tldahlgren committed:
tests/hdf5: convert to new stand-alone test process (#35732)

manuelakuhn committed:
py-cryptography: add 40.0.2 (#37925)

* py-cryptography: add 40.0.2

* Add pkgconfig dependency

manuelakuhn committed:
py-datalad-metalad: add 0-4-17 and new dep py-datalad-deprecated (#37993)

* py-datalad-metalad: add 0-4-17 and new dep py-datalad-deprecated

* Fix style

manuelakuhn committed:
py-datalad-container: add 1.2.0 (#38000)

manuelakuhn committed:
py-duecredit: add 0.9.2 (#38002)

manuelakuhn committed:
py-dunamai: add 1.17.0 (#38003)

michaelkuhn committed:
gobject-introspection: add 1.76.1 (#37995)

tmadlener committed:
podio: Add 0.16.5 tag (#37994)

zzzoom committed:
ucx: add 1.14.1 (#37991)

michaelkuhn committed:
julia: fix build for @1.8.4:1.8.5 (#37990)

julia@1.8.4:1.8.5 fails to build because it does not find libstdc++ (see https://github.com/JuliaLang/julia/issues/47987).

berquist committed:
Update SST packages to 13.0.0 (#37467)

* add sst-{core,elements,macro} v13.0.0

* Add newest DUMPI versions and remove unavailable ones

* update maintainer lists

* sst-core: tracking and profiling flags

* sst-elements with Pin requires the Pin location

* sst-core: Zoltan integration was removed in version 12

* spack style fixes

* sst-core: ensure Python is in the sst{sim,info}.x rpaths

* sst-macro: update homepage and maintainers

* spack style --fix

jack-morrison committed:
intel-mpi-benchmarks: Add MPI implementation check variant (#37363)

* Add MPI implementation check variant to Intel MPI Benchmarks

* [@spackbot] updating style on behalf of jack-morrison

alalazo committed:
Simplify implementation of "get_compiler_config" (#37989)

amd-toolchain-support committed:
cp2k fixes for aocc (#37758)

amd-toolchain-support committed:
HPL: amdalloc with AOCC 4.0 (#37757)

* adding amdalloc when using aocc 4


* adding and libm for aocc3.2

dev-zero committed:
Bugfix: cray manifest parsing regression (#37909)

fa7719a changed syntax for specifying exact versions, which are
required for some compiler specs (including those read as part
of parsing a Cray manifest). This fixes that and also makes a
couple other improvements to manifest parsing.

* Instantiate compiler specs with exact versions (fixes #37893)
* fix slingshot network detection (CPE 22.10+ has libcxi.so
  in /usr/lib64)
* "spack external find": add arg to ignore default dir for cray
  manifests

adamjstewart committed:
py-pandas: add v2.0.2 (#38005)

michaelkuhn committed:
gcc: add 11.4.0 (#37988)

MatthewLieber committed:
Merge branch 'OSU/hwloc-gpu' of https://github.com/OSU-Nowlab/spack into OSU/hwloc-gpu

alalazo commented:
Closing (the PR seems to have been force pushed in a wrong way, and never fixed)
